megamek/src/megamek/server/Server.java

```diff
@@ -887,6 +887,16 @@ private void receivePlayerVersion(Packet packet, int connId) {
             System.out.println("ERROR: Client/Server Version Mismatch -- Client: "+version+" Server: "+MegaMek.VERSION);
             needs = true;
         }
+        else if (clientChecksum == null) {    // print a message indicating client doesn't have jar file
```

Comment:
You need to add code like this:
```
if (!version.equals(MegaMek.VERSION)) {
                 buf.append(System.lineSeparator());
}
```
What this is doing is adding a new line.  Otherwise, if you have a version mismatch **and** checksum mismatch, the two error messages will run together.  The new line will keep them on a separate line.